superset-frontend/packages/superset-ui-chart-controls/src/components/ColumnOption.tsx

```diff
@@ -48,14 +52,27 @@ export function ColumnOption({
   const hasExpression = expression && expression !== column_name;
   const type = hasExpression ? 'expression' : type_generic;
   const [tooltipText, setTooltipText] = useState<ReactNode>(column.column_name);
+  const [columnTypeTooltipText, setcolumnTypeTooltipText] = useState<ReactNode>(
+    column.type,
+  );
 
   useLayoutEffect(() => {
     setTooltipText(getColumnTooltipNode(column, labelRef));
+    setcolumnTypeTooltipText(getColumnTypeTooltipNode(column));
   }, [labelRef, column]);
 
   return (
     <StyleOverrides>
-      {showType && type !== undefined && <ColumnTypeLabel type={type} />}
+      <Tooltip
```

Comment:
Just confirming something, since I can't easily test this at the moment... I see the tooltip is always rendering, whereas the _contents_ are conditionally rendered. If the column type is undefined, I mainly want to make sure we don't get an empty tooltip. But I'm actually wondering if we should just conditionally render the Tooltip anyway, for cleanliness ¯\\\_(ツ)_/¯

Reply:
- content ColumnTypeLabel is rendred based on type which is type_generic like NUMERIC, STRING, TEMPORAL, BOOLEAN etc. ColumnTypeLabel is rendered (with function type icon) even when type_generic is blank. Please refer below code.

```
export function ColumnTypeLabel({ type }: ColumnTypeLabelProps) {
  let typeIcon: ReactNode = (
    <QuestionOutlined aria-label={t('unknown type icon')} />
  );

  if (type === '' || type === 'expression') {
    typeIcon = <FunctionSvg aria-label={t('function type icon')} />;
  } else if (type === GenericDataType.STRING) {
    typeIcon = <StringSvg aria-label={t('string type icon')} />;
  } 
```

- where as tooltip content is rendered based on actual type which is TEXT, STRING, DOUBLE PRECISION etc. if type is undefined, null or blank then tooltip will not render anything (no extra UI component created as such cases are handled inside Tooltip component) as _getColumnTypeTooltipNode_ method will return null in such case. So, it is already conditional and I have verified such cases also in my local setup.